Config: update schemas, fix paginations

ol-zayatsm opened the pull request:
This PR introduces a set of improvements and refactors to the configuration system, focusing on schema cleanup, standardization of documentation and paginations fix. Key changes include:

      - Add extend_schema for basic method
      
      - Standardize API documentation structure
      
      - Improve parameter descriptions and examples
      
      - Ensure all endpoints are properly tagged under 'Configuration'
      
      - Add error response schemas for consistency

      - Remove unhandled paginations

ol-nata commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -20,8 +20,27 @@
 from bublik.data.serializers import ConfigSerializer
 
+from drf_spectacular.utils import (
+    extend_schema, 
```
You left some extra spaces here and elsewhere. Use the pyformat script to format the code and check your changes: https://github.com/ts-factory/bublik?tab=readme-ov-file#development

ol-nata commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -34,14 +53,202 @@ class Meta:
             },
         }
 
+# Serialzers for documentation of responses
```
Don’t mix the API and serialization. We have a separate folder for serializers.

ol-nata reviewed (changes requested):
Let’s start by fixing the following.

ol-nata commented on bublik/data/serializers/docs.py:
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2024 OKTET Labs Ltd. All rights reserved.
```
If you’re creating a new file, the copyright notice should use the current year.

ol-nata commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -35,13 +58,185 @@ class Meta:
+        description='''
+        Creates new configuration.
+        Requires adminisrtator's role.
+        The configuration will be created if there is no configuration with the same name.
```
Replace with: "The configuration is created only if the project does not already contain a configuration with the same type-name."

ol-nata commented on bublik/data/serializers/docs/config.py:
```diff
@@ -0,0 +1,17 @@
+    type = serializers.CharField()
+    name = serializers.CharField()
+    project = serializers.IntegerField()
+    all_config_versions = serializers.ListField(child=ConfigSerializer())
```
The fields from `ConfigSerializer` do not match the fields actually returned by the API.

ol-nata commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -169,12 +370,33 @@ def partial_update(self, request, *args, **kwargs):
+        ],
+        responses={
+            200: 'Successful receipt of the JSON schema',
+            422: 'Invalid configuration type or name',
```
1. Description?
2. The data format in the example response is too abstract.

ol-zayatsm replied:
Fixed.

ol-zayatsm commented on bublik/data/serializers/docs/config.py:
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
```
Fixed.

ol-nata reviewed (changes requested):
Overall, please reorganize the commits so that each documentation fix per method is a separate commit. This will make the changes easier to review and track.

ol-zayatsm committed:
config: remove pagination from ViewSet for consistency

Remove pagination from ConfigViewSet to align the generated list
documentation with the actual API behavior.

Signed-off-by: Mikhail Zayats <mikhail.zayats@oktetlabs.ru>

ol-nata commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -35,7 +54,94 @@ class Meta:
+        Updates a configuration or creates a new version.
+        Requires administrator's role.
+        ''',
+        request=ConfigSerializer,
```
The `type` field in `partial_update()` is not processed. Please use another serializer so the request example reflects reality.

ol-zayatsm replied:
Fixed.

ol-zayatsm commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -35,13 +59,201 @@ class Meta:
+        ],
+        responses={
+            200: OpenApiResponse(
+                response=ConfigSerializer(many=True),
```
Fixed.

ol-nata commented on bublik/interfaces/api_v2_docs/config/serializers.py:
```diff
@@ -0,0 +1,53 @@
+    config_types_names = ConfigTypeNameSerializer(many=True)
+
+
+class ErrorResponseSerializer(serializers.Serializer):
```
This serializer will be used not only for configurations. Please move it to a separate *exceptions.py* file inside the *api_v2_docs* directory.

Also, starting from version v2.9.0, error handling has been unified. Exception responses are now generated by `custom_exception_handler()`.

Please align the serializer structure with the format produced by `custom_exception_handler()`, including field names and their data types.

ol-zayatsm replied:
Fixed.

ol-zayatsm commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -75,12 +86,16 @@ def get_queryset(self):
 
         return configs.none()
 
+    @auth_required()
```
Fixed.

ol-zayatsm commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -75,12 +86,16 @@ def get_queryset(self):
+        return super().retrieve(request, *args, **kwargs)
+
+    @auth_required(as_admin=True)
+    def update(self, request, *args, **kwargs):
```
Fixed.

ol-zayatsm commented on bublik/interfaces/api_v2/config.py:
```diff
@@ -20,9 +20,14 @@
+from bublik.interfaces.api_v2_docs.config.schemas import config_viewset_schema
 
 
 class ConfigFilterSet(filters.FilterSet):
```
Fixed.

ol-zayatsm committed:
config: fix list method ignoring query parameter filters

Wire up filter_queryset so that query parameters are actually
applied to the queryset in the list method. Also explicitly
materialize the queryset via list() before passing it to
Response.

Signed-off-by: Mikhail Zayats <mikhail.zayats@oktetlabs.ru>

ol-zayatsm committed:
config: clarify OpenAPI response descriptions

Add explicit descriptions for all configuration endpoint responses,
including successful and error cases, so that the generated OpenAPI
schema documents the API behavior consistently and is easier to read.

Signed-off-by: Mikhail Zayats <mikhail.zayats@oktetlabs.ru>
